src/cloudai/_core/base_runner.py

```diff
@@ -287,11 +288,36 @@ async def monitor_jobs(self) -> int:
         self.logger.debug("Monitoring jobs.")
         for job in list(self.jobs):
             if self.is_job_completed(job):
-                completed_jobs_count += 1
-                await self.handle_job_completion(job)
+                if self.mode == "dry-run":
```

Comment:
Would be nice to somehow use the default no-op behaviour here to avoid this conditions.

Reply:
I didn't quite understand. Could you clarify your suggestion?

Reply:
@amaslenn  No-op = run = else part, no?